File: AnkiDroid/src/main/java/com/ichi2/utils/HtmlUtils.java
package com.ichi2.utils;

import android.text.TextUtils;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class HtmlUtils {
    //#5188 - compat.fromHtml converts newlines into spaces.
    @Nullable
    public static String convertNewlinesToHtml(@Nullable String html) {
        if (html == null) {
            return null;
        }
        String withoutWindowsLineEndings = html.replace("\r\n", "<br/>");
        //replace unix line endings
        return withoutWindowsLineEndings.replace("\n", "<br/>");
    }


    @NonNull
    public static String escape(@NonNull String html) {
        return TextUtils.htmlEncode(html);
    }
}

File: AnkiDroid/src/main/java/com/ichi2/utils/HtmlUtils.kt
// SPDX-License-Identifier: GPL-3.0-or-later

package com.ichi2.utils

import androidx.core.text.HtmlCompat

/** Removes HTML tags from a string */
fun stripHtml(html: String): String = HtmlCompat.fromHtml(html, HtmlCompat.FROM_HTML_MODE_LEGACY).toString()
